lib/resources/rabbitmq_conf.rb

```diff
@@ -0,0 +1,68 @@
+# encoding: utf-8
+# author: Dominik Richter
+# author: Christoph Hartmann
+
+require 'utils/erlang_parser'
+
+module Inspec::Resources
+  class RabbitmqConf < Inspec.resource(1)
+    name 'rabbitmq_config'
+    desc 'Use the rabbitmq_config InSpec resource to test configuration data '\
+         'for the RabbitMQ service located in /etc/rabbitmq/rabbitmq.config on '\
+         'Linux and UNIX platforms.'
+    example "
+      describe rabbitmq_config.params('rabbit', 'ssl_listeners') do
+        it { should cmp 5671 }
+      end
+    "
+
+    def initialize(conf_path = nil)
+      @conf_path = conf_path || '/etc/rabbitmq/rabbitmq.config'
+    end
+
+    def params(*opts)
+      opts.inject(read_params) do |res, nxt|
+        res.respond_to?(:key) ? res[nxt] : nil
+      end
+    end
+
+    def to_s
+      'RabbitMQ Configuration'
+    end
+
+    private
+
+    def read_content
+      return @content if defined?(@content)
```

Comment:
The reason I started to use this convention initially was to avoid queries to the system for files that might be `nil`. It typically happens with missing permissions, permission the file read may be attempted a large number of times. It is only meant as an optimization.

That said, this part as well as the pieces below really need to be abstracted away, since these steps are used frequently.